pyo3-build-config/src/impl_.rs

```diff
@@ -709,6 +766,44 @@ impl CrossCompileEnvVars {
             || self.pyo3_cross_lib_dir.is_some()
             || self.pyo3_cross_python_version.is_some()
     }
+
+    /// Parses `PYO3_CROSS_PYTHON_VERSION` environment variable value
+    /// into `PythonVersion`.
+    fn parse_version(&self) -> Result<Option<PythonVersion>> {
+        let version = self
+            .pyo3_cross_python_version
+            .as_ref()
+            .map(|os_string| {
+                let utf8_str = os_string
+                    .to_str()
+                    .ok_or("PYO3_CROSS_PYTHON_VERSION is not valid a UTF-8 string")?;
+                utf8_str
+                    .parse()
+                    .context("failed to parse PYO3_CROSS_PYTHON_VERSION")
+            })
+            .transpose()?;
+
+        Ok(version)
+    }
+
+    /// Converts the stored `PYO3_CROSS_LIB_DIR` variable value (if any)
+    /// into a `PathBuf` instance.
+    ///
+    /// Ensures that the path is a valid UTF-8 string.
+    fn lib_dir_path(&self) -> Result<PathBuf> {
+        let lib_dir = self.pyo3_cross_lib_dir.as_ref().map(PathBuf::from);
+
+        if let Some(dir) = lib_dir.as_ref() {
+            if dir.to_str().is_none() {
```

Comment:
This look like the `ensure!` macro could improve readability.

Reply:
Applied in https://github.com/PyO3/pyo3/commit/9b45a19161cd20fac1c83c25897aa9d213244a15